sympy/functions/elementary/hyperbolic.py

```diff
@@ -926,6 +927,21 @@ def eval(cls, arg):
                 if _coeff_isneg(arg):
                     return -cls(-arg)
 
+        if isinstance(arg, sinh):
+            x = arg.args[0]
+            if x.is_number:
+                if x.is_real:
+                    return x
+                r, i = match_real_imag(x)
+                if r is not None and i is not None:
+                    f = floor((i + pi/2)/pi)
+                    if f.is_number and f.is_integer:
+                        m = r + I*i - I*pi*f
+                        if f % 2 == 0:
+                            return m
```

Comment:
We already know that f.is_number because f comes from x and we checked that x.is_number. Likewise if f is floor(...) then f.is_integer *should* always be True (although it might sometimes return None). It is still possible though that f.is_number and f.is_integer but we can't calculate f%2. I would say that the check would be better as:
```python
even = f.is_even
if even is True:
    return m
elif even is False:
    return -m
# Note the 3rd case even is None where we don't evaluate
```